thingsearch/service/src/main/java/org/eclipse/ditto/thingsearch/service/persistence/write/impl/MongoThingsSearchUpdaterPersistence.java

```diff
@@ -106,18 +110,26 @@ public Source<PolicyReferenceTag, NotUsed> getPolicyReferenceTags(final Map<Poli
                         .append(PersistenceConstants.FIELD_POLICY_ID, new BsonInt32(1))
                         .append(PersistenceConstants.FIELD_REFERENCED_POLICIES, new BsonInt32(1)));
 
-        return Source.fromPublisher(publisher)
-                .mapConcat(doc -> {
-                    final ThingId thingId = ThingId.of(doc.getString(PersistenceConstants.FIELD_ID));
-                    final Collection<PolicyId> referencedPolicyIds = referencedPolicyIds(doc);
-                    return referencedPolicyIds.stream()
-                            .map(referencedPolicyId -> Optional.ofNullable(policyRevisions.get(referencedPolicyId))
-                                    .map(revision -> PolicyTag.of(referencedPolicyId, revision))
-                                    .map(policyTag -> PolicyReferenceTag.of(thingId, policyTag))
-                                    .orElse(null))
-                            .filter(Objects::nonNull)
-                            .toList();
-                });
+        var source = Source.fromPublisher(publisher);
```

Comment:
I would rather do this not as mutable source - but make it final instead.
Meaning adding e.g. an `else` where you would only initialize `source = Source.fromPublisher(publisher)` - and when throttling enabled, do in addition the `.throttle()`